src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.INumericTC.cs

```diff
@@ -36,20 +38,21 @@ private interface INumericTC<T>
             T MaxValue { get; }
 
             /// <summary>
-            /// Returns the midpoints when subdividing an interval, which becomes the
-            /// left subinterval's max value and the right subinterval's min value.
-            /// The returned rightMin must be the successor value (see <see cref="Next(T)"/>) to leftMax.
+            /// The successor (next larger) value to a given value. The result is not defined
+            /// when <paramref name="value"/> is <see cref="MaxValue"/>.
             /// </summary>
-            /// <param name="min">the parent interval's minimum value, inclusive.</param>
-            /// <param name="max">the parent interval's maximum value, inclusive</param>
-            (T leftMax, T rightMin) Partition(T min, T max);
+            T Next(T value);
 
             /// <summary>
-            /// The successor (next larger) value to a given value. Used to determine when two intervals
-            /// are contiguous to improve the output of <see cref="object.ToString"/>. The result is not defined
-            /// when <paramref name="value"/> is <see cref="MaxValue"/>.
+            /// The predecessor (previous larger) value to a given value. The result is not defined
+            /// when <paramref name="value"/> is <see cref="MinValue"/>.
             /// </summary>
-            T Next(T value);
+            T Prev(T value);
+
+            /// <summary>
+            /// Produce a randomly-selected value.
+            /// </summary>
+            T Random(Random random);
```

Comment:
I'm curious what the value of this is.

Reply:
It is a building block for making a random value set which is used for testing.

Reply:
ah, might be good to doc that.  from just reading it, it seemed super strange (given our desire for determinism).  For tests though I can see why this would be helpful :)